### PR TITLE
Add test with both pthread and newlib enabled

### PR DESCRIPTION
**Pull request by vanti**
This commit adds a test case that reproduces sdk-ng#45.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

### CODE REVIEW

File: tests/posix/newlib/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+tests:
+  portability.posix:
+    tags: posix
+    min_ram: 64
```
Comment (galak) [inline tests/posix/newlib/testcase.yaml]: You should exclude native_posix

Comment (galak) [reply tests/posix/newlib/testcase.yaml]: and the toolchain supporting newlib:

`filter: TOOLCHAIN_HAS_NEWLIB == 1`

Comment (vanti) [reply tests/posix/newlib/testcase.yaml]: Thanks, updated.

**Review by pfalcon — APPROVED** [review]

**Comment by galak** [comment]
Closing this in favor of #13685

**Pull request closed without merging**